### PR TITLE
[ntfs2gtfs] write trips.txt

### PR DESCRIPTION
**Pull request by datanel**
see https://github.com/CanalTP/navitia_model/blob/master/src/documentation/ntfs2gtfs.md#tripstxt

### CODE REVIEW

File: src/gtfs/write.rs [hunk 1]
```diff
@@ -58,6 +62,91 @@ pub fn write_stops(
...
+    }
+}
+
+fn get_gtfs_direction_id_from_ntfs_vj(
+    vj: &objects::VehicleJourney,
+    routes: &CollectionWithId<objects::Route>,
+) -> DirectionType {
+    let route = routes.get(&vj.route_id).unwrap();
+    match route.direction_type.as_ref().map(|s| s.as_str()) {
+        Some("foward") | Some("clockwise") | Some("inbound") => DirectionType::Backward,
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/gtfs/write.rs]: be carefull of the type : "foward" => "fo**r**ward"

File: src/gtfs/write.rs [hunk 2]
```diff
@@ -58,6 +62,91 @@ pub fn write_stops(
     Ok(())
 }
 
+fn get_gtfs_trip_shortname_and_headsign_from_ntfs_vj(
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/gtfs/write.rs]: We discussed this point, the need is to split the NTFS `trip_headsign` in the GTFS 2 available values `trip_short_name` and `trip_headsign`

File: src/gtfs/write.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (datanel) [reply src/gtfs/write.rs]: it was an error in the specs

File: src/gtfs/write.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (datanel) [reply src/gtfs/write.rs]: > Can I have the spec of that?

have you read the specs? it it ok for you?

File: src/gtfs/write.rs [hunk 3]
```diff
@@ -58,6 +62,91 @@ pub fn write_stops(
...
+) -> (Option<String>, Option<String>) {
+    fn get_last_stop_name(
+        vj: &objects::VehicleJourney,
+        sps: &CollectionWithId<objects::StopPoint>,
+    ) -> Option<String> {
+        vj.stop_times
+            .iter()
+            .max_by_key(|st| st.sequence)
+            .map(|st| &sps[st.stop_point_idx].name)
+            .cloned()
```
[the remark this answers is not in the document]
Comment (datanel) [reply src/gtfs/write.rs]: `back` does not exist, I used `last` and without `cloned` I have an `Option<&String>` not `Option<String>`

[the remark this answers is not in the document]
Comment (TeXitoi) [reply src/gtfs/write.rs]: I've just added the clone in the map, but that the same.

Yes, I mean last (back is c++)

File: src/gtfs/write.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TeXitoi) [reply src/gtfs/write.rs]: Yes, nothing to note about the PR, just that maybe the specs might change, but it involve fusio and navitia, so out of this scope.

File: src/gtfs/mod.rs [hunk 4]
```diff
@@ -145,7 +145,7 @@ impl<'a> From<&'a objects::StopArea> for Stop {
 }
 
 #[derive(Derivative)]
-#[derivative(Default(bound = ""))]
+#[derivative(Default)]
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
```
Comment (TeXitoi) [inline src/gtfs/mod.rs]: I mean simply no derivative:
```rust
#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash, Default)]
```

**Pull request merged**